src/redux/loaderSlice.tsx

```diff
@@ -0,0 +1,38 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit"
+
+export interface ILoaderState {
+    isLoading: boolean,
+    counter:number,
+
+}
+const initialState: ILoaderState = {
+    isLoading: false,
+    counter:0,
+
+}
+
+export const loaderSlice = createSlice({
+    name: 'loaderReducer',
+    initialState,
+    reducers: {
+        startLoader: (state: ILoaderState, action: PayloadAction<void>) => {
+            if(state.counter==0){
```

Comment:
check the condition only when stopLoading:
startLoading: 
isLoading = true;
counter++

stop loading :
counter--
isLoading = counter === 0